### ISSUE TITLE
parse_authority() incorrectly parses IPv6 addresses

### ISSUE THREAD

**Issue by nirbhayc** [codership/galera#338]
./bin/mysqld --wsrep-node-address=fe80::a00:27ff:fe0b:c0b0

```
2015-09-03 12:40:31 140143098599360 [ERROR] WSREP: Can't parse port number from 'c0b0': 22 (Invalid argument)
     at galerautils/src/gu_uri.cpp:parse_authority():69
2015-09-03 12:40:31 140143098599360 [ERROR] WSREP: wsrep::init() failed: 7, must shutdown
2015-09-03 12:40:31 140143098599360 [ERROR] Aborting
```

**Comment by nirbhayc** [comment]
The rule/assumption that port always follows the last colon isn't correct for IPv6.

https://github.com/codership/galera/blob/3.x/galerautils/src/gu_uri.cpp#L47

**Comment by temeo** [comment]
IPv6 addresses should be enclosed by brackets `[ ]`.

**Comment by nirbhayc** [comment]
Not unless there is a port.

**Comment by nirbhayc** [comment]
```
2015-09-03 12:50:01 139687971100608 [ERROR] WSREP: Can't parse port number from 'c0b0]': 22 (Invalid argument)
     at galerautils/src/gu_uri.cpp:parse_authority():69

```

**Comment by wido** [comment]
Is there any progress on this? I would like to run a MariaDB Galera cluster in a IPv6-only network.

**Comment by ayurchen** [comment]
you can perfectly do that. just use the host names instead of IPv6 format.

**Comment by temeo** [comment]
Should be fixed by c6ada002813093ab56e387ebe237a45600bbd775
